<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CPathSettingsDialog</class>
 <widget class="QDialog" name="CPathSettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>570</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Path Generator Settings</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QDialog {
    background-color: rgb(30, 30, 40);
    color: white;
}
QLabel {
    color: white;
    font-weight: bold;
    font-size: 10pt;
}
QGroupBox {
    color: #FFB74D;
    font-weight: bold;
    font-size: 11pt;
    border: 2px solid rgba(255, 152, 0, 0.4);
    border-radius: 8px;
    margin-top: 12px;
    padding-top: 10px;
}
QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top left;
    padding: 0 8px;
    left: 10px;
}
QSpinBox, QDoubleSpinBox {
    background-color: rgb(50, 50, 65);
    color: white;
    border: 1px solid rgba(255, 152, 0, 0.5);
    border-radius: 4px;
    padding: 5px;
    font-size: 10pt;
}
QSpinBox:focus, QDoubleSpinBox:focus {
    border: 2px solid rgba(255, 167, 38, 0.8);
}
QPushButton {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 rgba(255, 152, 0, 0.85),
                                stop:0.5 rgba(245, 124, 0, 0.85),
                                stop:1 rgba(230, 81, 0, 0.85));
    border: 2px solid rgba(255, 152, 0, 0.4);
    color: white;
    padding: 10px 25px;
    border-radius: 8px;
    font-weight: 600;
    font-size: 10pt;
}
QPushButton:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 rgba(255, 167, 38, 0.9),
                                stop:0.5 rgba(255, 152, 0, 0.9),
                                stop:1 rgba(245, 124, 0, 0.9));
    border: 2px solid rgba(255, 167, 38, 0.6);
}
QPushButton:pressed {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 rgba(230, 81, 0, 0.9),
                                stop:1 rgba(191, 54, 12, 0.9));
}
QPushButton#pushButton_ResetDefaults {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 rgba(97, 97, 97, 0.85),
                                stop:1 rgba(66, 66, 66, 0.85));
    border: 2px solid rgba(150, 150, 150, 0.4);
}
QPushButton#pushButton_ResetDefaults:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 rgba(117, 117, 117, 0.9),
                                stop:1 rgba(97, 97, 97, 0.9));
}
QPushButton#pushButton_Cancel {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 rgba(97, 97, 97, 0.85),
                                stop:1 rgba(66, 66, 66, 0.85));
    border: 2px solid rgba(150, 150, 150, 0.4);
}
QPushButton#pushButton_Cancel:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                stop:0 rgba(117, 117, 117, 0.9),
                                stop:1 rgba(97, 97, 97, 0.9));
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox_General">
     <property name="title">
      <string>General Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="horizontalSpacing">
       <number>15</number>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>15</number>
      </property>
      <property name="bottomMargin">
       <number>15</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_NumWaypoints">
        <property name="text">
         <string>Number of Waypoints:</string>
        </property>
        <property name="toolTip">
         <string>Number of intermediate waypoints to generate in the path (2-100)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="spinBox_NumWaypoints">
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_Altitude">
        <property name="text">
         <string>Default Altitude (m):</string>
        </property>
        <property name="toolTip">
         <string>Default altitude in meters for generated waypoints</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_Altitude">
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>50000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>100.000000000000000</double>
        </property>
        <property name="value">
         <double>1000.000000000000000</double>
        </property>
        <property name="suffix">
         <string> m</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_Curve">
     <property name="title">
      <string>Curve Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="horizontalSpacing">
       <number>15</number>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>15</number>
      </property>
      <property name="bottomMargin">
       <number>15</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_CurveFactor">
        <property name="text">
         <string>Curve Factor:</string>
        </property>
        <property name="toolTip">
         <string>Controls curve intensity (0.0 = straight, 1.0 = maximum curve)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_CurveFactor">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.300000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_SpiralTurns">
        <property name="text">
         <string>Spiral Turns:</string>
        </property>
        <property name="toolTip">
         <string>Number of spiral rotations for spiral path type (0.5-10)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_SpiralTurns">
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="minimum">
         <double>0.500000000000000</double>
        </property>
        <property name="maximum">
         <double>10.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.500000000000000</double>
        </property>
        <property name="value">
         <double>2.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_MaxTurnRadius">
        <property name="text">
         <string>Max Turn Radius (°):</string>
        </property>
        <property name="toolTip">
         <string>Maximum turn radius in degrees for L-curve paths</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_MaxTurnRadius">
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>0.200000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.050000000000000</double>
        </property>
        <property name="suffix">
         <string>°</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_SpreadRadius">
        <property name="text">
         <string>Spread Radius (km):</string>
        </property>
        <property name="toolTip">
         <string>Maximum lateral spread of the trajectory from the main path in kilometers</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_SpreadRadius">
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="minimum">
         <double>0.500000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>5.000000000000000</double>
        </property>
        <property name="suffix">
         <string> km</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_Zigzag">
     <property name="title">
      <string>Zigzag Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="horizontalSpacing">
       <number>15</number>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>15</number>
      </property>
      <property name="bottomMargin">
       <number>15</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_ZigzagAmplitude">
        <property name="text">
         <string>Zigzag Amplitude:</string>
        </property>
        <property name="toolTip">
         <string>Amplitude of zigzag as fraction of total distance (0.05-0.5)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_ZigzagAmplitude">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0.050000000000000</double>
        </property>
        <property name="maximum">
         <double>0.500000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.200000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_ZigzagFrequency">
        <property name="text">
         <string>Zigzag Frequency:</string>
        </property>
        <property name="toolTip">
         <string>Number of zigzag oscillations (2-20)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinBox_ZigzagFrequency">
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="maximum">
         <number>20</number>
        </property>
        <property name="value">
         <number>6</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_Random">
     <property name="title">
      <string>Random Path Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <property name="horizontalSpacing">
       <number>15</number>
      </property>
      <property name="verticalSpacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>15</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>15</number>
      </property>
      <property name="bottomMargin">
       <number>15</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_RandomVariance">
        <property name="text">
         <string>Random Variance:</string>
        </property>
        <property name="toolTip">
         <string>Variance factor for random path generation (0.05-0.5)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinBox_RandomVariance">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0.050000000000000</double>
        </property>
        <property name="maximum">
         <double>0.500000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.150000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_Buttons">
     <property name="spacing">
      <number>15</number>
     </property>
     <item>
      <widget class="QPushButton" name="pushButton_ResetDefaults">
       <property name="text">
        <string>Reset Defaults</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_Cancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_Apply">
       <property name="text">
        <string>Apply</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
